### ISSUE TITLE
unable to access app after docker compose

### ISSUE THREAD

**Issue by Awawdi** [docker/getting-started#86]
I followed instruction one by one, once containers were composed using 
docker run -d --network todo-app --network-alias mysql -v todo-mysql-data:/var/lib/mysql -e MYSQL_ROOT_PASSWORD=secret -e MYSQL_DATABASE=todos mysql:5.7

I am unable to login into app using http://localhost:3000

terminal:

Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

PS C:\Users\LenovoX\Downloads\app\app> docker build -t getting-started .
Sending build context to Docker daemon  4.689MB
Error response from daemon: the Dockerfile (Dockerfile) cannot be empty
PS C:\Users\LenovoX\Downloads\app\app> docker build -t getting-started .
Sending build context to Docker daemon   4.69MB
Step 1/5 : FROM node:12-alpine
 ---> 18f4bc975732
Step 2/5 : WORKDIR /app
 ---> Using cache
 ---> 10872044b09c
Step 3/5 : COPY . .
 ---> 0f397df0b22e
Step 4/5 : RUN yarn install --production
 ---> Running in 6db94dc03cad
yarn install v1.22.4
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@1.2.9: The platform "linux" is incompatible with this module.
info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Removing intermediate container 6db94dc03cad
Step 5/5 : CMD ["node", "src/index.js"]
 ---> Running in dff25f6bd75c
Removing intermediate container dff25f6bd75c
 ---> c3f583ac3818
Successfully built c3f583ac3818
Successfully tagged getting-started:latest
SECURITY WARNING: You are building a Docker image from Windows against a non-Windows Docker host. All files and directories added to build context will have '-rwxr-xr-x' permissions. It is recommended to double check and reset permissions for sensitive files and directories.
PS C:\Users\LenovoX\Downloads\app\app> docker run -dp 3000:3000 getting-started
9637c26796f3d2dc1128df885dafb1386defeda26bc7f69cd6d6fbe38d0754c3
PS C:\Users\LenovoX\Downloads\app\app> docker build -t getting-started .
Sending build context to Docker daemon   4.69MB
Step 1/5 : FROM node:12-alpine
 ---> 18f4bc975732
Step 2/5 : WORKDIR /app
 ---> Using cache
 ---> 10872044b09c
Step 3/5 : COPY . .
 ---> 34b658d040a8
Step 4/5 : RUN yarn install --production
 ---> Running in cd0f518a7dd0
yarn install v1.22.4
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@1.2.9: The platform "linux" is incompatible with this module.
info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Removing intermediate container cd0f518a7dd0
 ---> 6e77a025cf17
Step 5/5 : CMD ["node", "src/index.js"]
Removing intermediate container 15a722b004e0
Successfully built 737e4158a1b6
Successfully tagged getting-started:latest
t permissions for sensitive files and directories.
PS C:\Users\LenovoX\Downloads\app\app> docker run -dp 3000:3000 getting-started
y allocated.
PS C:\Users\LenovoX\Downloads\app\app> docker run -dp 3000:3000 getting-started
0e842a25e00a3353877f3ba6d50dcb16b887acec7d98f958e64c23335c230991
PS C:\Users\LenovoX\Downloads\app\app> docker login -u orsanaw .
Password:
Error response from daemon: Get https://./v2/: dial tcp: lookup .: no such host
PS C:\Users\LenovoX\Downloads\app\app> docker login -u orsanaw  
Password:
Login Succeeded
PS C:\Users\LenovoX\Downloads\app\app> docker push orsanaw/simple-task-list
47f92a3033a9: Pushed
02f5ff2722b6: Pushed
2f1a3c8394a7: Pushed
db809908a198: Mounted from library/node
1b235e8e7bda: Mounted from library/node
latest: digest: sha256:a007549c619786e14dd5b85d2cd3b7a96640a6d010dbcc31ab41429b72b41c59 size: 1788
PS C:\Users\LenovoX\Downloads\app\app> docker volume create todo-db
todo-db
Unable to find image 'simple-task-list:latest' locally
See 'docker run --help'.
Password:
PS C:\Users\LenovoX\Downloads\app\app> docker run -dp 3000:3000 -v todo-db:/etc/todos simple-task-list
Unable to find image 'simple-task-list:latest' locally
See 'docker run --help'.
PS C:\Users\LenovoX\Downloads\app\app> docker run -dp 3000:3000 -v todo-db:/etc/todos getting-started
5e33afafc7b596adf69cef0bb8312f80e8cf42a644cc6635d18688257169d64e
PS C:\Users\LenovoX\Downloads\app\app> docker run -dp 3000:3000 -v todo-db:/etc/todos getting-started
07e2bf930f133a50033626f3fb3d1d9034961dd7119c9f87e7ee30dff6d5e4ae
PS C:\Users\LenovoX\Downloads\app\app> docker network create todo-app
26b72f8e2c80abd6ff7015881afd21019433e60c6c4e5761ed37e5463483e9d9
PS C:\Users\LenovoX\Downloads\app\app> docker run -d \
docker: invalid reference format.
See 'docker run --help'.
PS C:\Users\LenovoX\Downloads\app\app>     --network todo-app --network-alias mysql \
+     --network todo-app --network-alias mysql \
+       ~
Missing expression after unary operator '--'.
At line:1 char:7
+     --network todo-app --network-alias mysql \
+       ~~~~~~~
Unexpected token 'network' in expression or statement.
    + FullyQualifiedErrorId : MissingExpressionAfterOperator
 
PS C:\Users\LenovoX\Downloads\app\app>     -v todo-mysql-data:/var/lib/mysql \
-v : The term '-v' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:5
+     -v todo-mysql-data:/var/lib/mysql \
+     ~~
    + FullyQualifiedErrorId : CommandNotFoundException
 
PS C:\Users\LenovoX\Downloads\app\app>     -e MYSQL_ROOT_PASSWORD=secret \
-e : The term '-e' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:5
+     -e MYSQL_ROOT_PASSWORD=secret \
+     ~~
    + FullyQualifiedErrorId : CommandNotFoundException
 
PS C:\Users\LenovoX\Downloads\app\app>     -e MYSQL_DATABASE=todos \
-e : The term '-e' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:5
+     -e MYSQL_DATABASE=todos \
+     ~~
    + CategoryInfo          : ObjectNotFound: (-e:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
PS C:\Users\LenovoX\Downloads\app\app> docker run -d --network todo-app --network-alias mysql -v todo-mysql-data:/var/lib/mysql -e MYSQL_ROOT_PASSWORD=secret -e MYSQL_DATABASE=todos mysql:5.7
Unable to find image 'mysql:5.7' locally
5.7: Pulling from library/mysql
d121f8d1c412: Pull complete
f3cebc0b4691: Pull complete
1862755a0b37: Pull complete
690874f836db: Pull complete
baa8be383ffb: Pull complete
277d8f888368: Pull complete
21f2da6feb67: Pull complete
2c98f818bcb9: Pull complete
031b0a770162: Pull complete
Status: Downloaded newer image for mysql:5.7
f29dec0db671c18efd3e86503bacd96b54ca52ec3db6e949da9c483da1add6cb
Enter password:
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
PS C:\Users\LenovoX\Downloads\app\app> docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
f29dec0db671        mysql:5.7           "docker-entrypoint.s…"   4 minutes ago       Up 4 minutes        3306/tcp, 33060/tcp      unruffled_morse
07e2bf930f13        getting-started     "docker-entrypoint.s…"   13 minutes ago      Up 13 minutes       0.0.0.0:3000->3000/tcp   boring_mayer
d78104b6cebd        docker101tutorial   "/docker-entrypoint.…"   4 hours ago         Up 4 hours          0.0.0.0:80->80/tcp       docker-tutial
PS C:\Users\LenovoX\Downloads\app\app> docker exec -it f29dec0db671 mysql -p
Enter password:
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: NO)
PS C:\Users\LenovoX\Downloads\app\app> docker exec -it f29dec0db671 mysql -p
Enter password: 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 4
Server version: 5.7.31 MySQL Community Server (GPL)

Copyright (c) 2000, 2020, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> SHOW DATABASES;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| mysql              |
| sys                |
| todos              |
+--------------------+
5 rows in set (0.00 sec)

mysql> docker run -it --network todo-app nicolaka/netshoot
    ->
    -> ^C
mysql> EXIT
Bye
PS C:\Users\LenovoX\Downloads\app\app> docker run -it --network todo-app nicolaka/netshoot
Unable to find image 'nicolaka/netshoot:latest' locally
latest: Pulling from nicolaka/netshoot
cbdbe7a5bc2a: Already exists
fa7edde5704a: Pull complete
d142e371ed28: Pull complete
7bc9cb006bce: Pull complete
a4d2c327d444: Pull complete
428e55c983a8: Pull complete
1209022df24d: Pull complete
b74093e72c31: Pull complete
Digest: sha256:04786602e5a9463f40da65aea06fe5a825425c7df53b307daa21f828cfe40bf8
Status: Downloaded newer image for nicolaka/netshoot:latest
                    dP            dP                           dP   
                    88            88                           88
88d888b. .d8888b. d8888P .d8888b. 88d888b. .d8888b. .d8888b. d8888P
88'  `88 88ooood8   88   Y8ooooo. 88'  `88 88'  `88 88'  `88   88
88    88 88.  ...   88         88 88    88 88.  .88 88.  .88   88
dP    dP `88888P'   dP   `88888P' dP    dP `88888P' `88888P'   dP

Welcome to Netshoot! (github.com/nicolaka/netshoot)
root @ / 
 [1] 🐳  → dig mysql

; <<>> DiG 9.14.12 <<>> mysql
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 24678
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;mysql.                         IN      A
;; ANSWER SECTION:
mysql.                  600     IN      A       172.18.0.2

;; Query time: 165 msec
;; SERVER: 127.0.0.11#53(127.0.0.11)
;; WHEN: Sun Sep 13 13:12:38 UTC 2020
;; MSG SIZE  rcvd: 44
docker: invalid reference format.
PS C:\Users\LenovoX\Downloads\app\app>   -w /app -v "$(pwd):/app" \
PS C:\Users\LenovoX\Downloads\app\app>   --network todo-app \
PS C:\Users\LenovoX\Downloads\app\app>   -e MYSQL_HOST=mysql \
-e : The term '-e' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
PS C:\Users\LenovoX\Downloads\app\app>   -e MYSQL_PASSWORD=secret \
-e : The term '-e' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:3
+   -e MYSQL_DB=todos \
+   ~~
    + CategoryInfo          : ObjectNotFound: (-e:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
PS C:\Users\LenovoX\Downloads\app\app>   node:12-alpine \
PS C:\Users\LenovoX\Downloads\app\app>   sh -c "yarn install && yarn run dev"
sh : The term 'sh' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:3
+   sh -c "yarn install && yarn run dev"
+   ~~
    + CategoryInfo          : ObjectNotFound: (sh:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
PS C:\Users\LenovoX\Downloads\app\app>
PS C:\Users\LenovoX\Downloads\app\app>
PS C:\Users\LenovoX\Downloads\app\app>
PS C:\Users\LenovoX\Downloads\app\app>
PS C:\Users\LenovoX\Downloads\app\app>
PS C:\Users\LenovoX\Downloads\app\app>
PS C:\Users\LenovoX\Downloads\app\app> docker run -dp 3000:3000 -w /app -v "$(pwd):/app" --network todo-app -e MYSQL_HOST=mysql -e MYSQL_USER=root -e MYSQL_PASSWORD=secret -e MYSQL_DB=todos node:12-alpine sh -c "yarn install 
&& yarn run dev"
17b0e853a37e26a3a8ea66209bdb76d137a6d3d973f06fcfad742e1731573864
docker: Error response from daemon: driver failed programming external connectivity on endpoint serene_wilbur (bfdc268e28eba09a10e7ea07dd2f4fa744caee77564f9a457d8bc1b7cbe465a6): Bind for 0.0.0.0:3000 failed: port is already allocated.
PS C:\Users\LenovoX\Downloads\app\app> docker run -dp 3000:3000 -w /app -v "$(pwd):/app" --network todo-app -e MYSQL_HOST=mysql -e MYSQL_USER=root -e MYSQL_PASSWORD=secret -e MYSQL_DB=todos node:12-alpine sh -c "yarn install 
&& yarn run dev"
cb90e35ebef2b9b58f3ebb7455c7b6eb41859d7cd33e6891a3ff744ed85303b4
Enter password:
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Your MySQL connection id is 7
Server version: 5.7.31 MySQL Community Server (GPL)

Copyright (c) 2000, 2020, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> select * from todo_items;
+--------------------------------------+------------+-----------+
| id                                   | name       | completed |
| 09eb8736-1b37-49e0-bd18-f92d895554b4 | buy Milk!  |         0 |
| ccff8405-86c5-492c-899e-47b394233e54 | buy Bread! |         0 |
+--------------------------------------+------------+-----------+
2 rows in set (0.00 sec)
mysql> docker-compose version
    -> ^C
mysql> ^C
mysql> exit
Bye
PS C:\Users\LenovoX\Downloads\app\app> docker-compose version
docker-py version: 4.2.2
CPython version: 3.7.4
OpenSSL version: OpenSSL 1.1.1c  28 May 2019
PS C:\Users\LenovoX\Downloads\app\app>
PS C:\Users\LenovoX\Downloads\app\app> docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
cb90e35ebef2        node:12-alpine      "docker-entrypoint.s…"   8 minutes ago       Up 8 minutes        0.0.0.0:3000->3000/tcp   competent_almeida
f29dec0db671        mysql:5.7           "docker-entrypoint.s…"   24 minutes ago      Up 24 minutes       3306/tcp, 33060/tcp      unruffled_morse
d78104b6cebd        docker101tutorial   "/docker-entrypoint.…"   4 hours ago         Up 4 hours          0.0.0.0:80->80/tcp       docker-tutial
PS C:\Users\LenovoX\Downloads\app\app> docker-compose up -d
ERROR: 
        Can't find a suitable configuration file in this directory or any

        Supported filenames: docker-compose.yml, docker-compose.yaml

PS C:\Users\LenovoX\Downloads\app\app> docker-compose up -d
Creating network "app_default" with the default driver
Creating app_app_1 ...
WARNING: Host is already in use by another container

ERROR: for app_app_1  Cannot start service app: driver failed programming external connectivity on endpoint app_app_1 (5868c601445ddfb694acfdc5b038920439fb8e4b664ab4a2a52ee7dbfb383d6f): Bind for 0.0.0.0:3000 failed: port is aCreating app_mysql_1 ... done
ERROR: for app  Cannot start service app: driver failed programming external connectivity on endpoint app_app_1 (5868c601445ddfb694acfdc5b038920439fb8e4b664ab4a2a52ee7dbfb383d6f): Bind for 0.0.0.0:3000 failed: port is alreadyERROR: Encountered errors while bringing up the project.
PS C:\Users\LenovoX\Downloads\app\app> docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED              STATUS              PORTS                    NAMES
cb90e35ebef2        node:12-alpine      "docker-entrypoint.s…"   11 minutes ago       Up 11 minutes       0.0.0.0:3000->3000/tcp   competent_almeida
d78104b6cebd        docker101tutorial   "/docker-entrypoint.…"   4 hours ago          Up 4 hours          0.0.0.0:80->80/tcp       docker-tutial
PS C:\Users\LenovoX\Downloads\app\app> docker rm -f 612dede4c20d
PS C:\Users\LenovoX\Downloads\app\app> docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
cb90e35ebef2        node:12-alpine      "docker-entrypoint.s…"   11 minutes ago      Up 11 minutes       0.0.0.0:3000->3000/tcp   competent_almeida
f29dec0db671        mysql:5.7           "docker-entrypoint.s…"   27 minutes ago      Up 27 minutes       3306/tcp, 33060/tcp      unruffled_morse
PS C:\Users\LenovoX\Downloads\app\app> docker rm -f f29dec0db671
f29dec0db671
PS C:\Users\LenovoX\Downloads\app\app> docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
cb90e35ebef2        node:12-alpine      "docker-entrypoint.s…"   12 minutes ago      Up 12 minutes       0.0.0.0:3000->3000/tcp   competent_almeida
d78104b6cebd        docker101tutorial   "/docker-entrypoint.…"   4 hours ago         Up 4 hours          0.0.0.0:80->80/tcp       docker-tutial
PS C:\Users\LenovoX\Downloads\app\app> docker rm -f cb90e35ebef2
cb90e35ebef2
PS C:\Users\LenovoX\Downloads\app\app> docker-compose up -d
Starting app_app_1   ... done
Creating app_mysql_1 ... done
PS C:\Users\LenovoX\Downloads\app\app> docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
39ce32aa8442        mysql:5.7           "docker-entrypoint.s…"   5 minutes ago       Up 5 minutes        3306/tcp, 33060/tcp      app_mysql_1
a2199bae223e        node:12-alpine      "docker-entrypoint.s…"   9 minutes ago       Up 5 minutes        0.0.0.0:3000->3000/tcp   app_app_1
d78104b6cebd        docker101tutorial   "/docker-entrypoint.…"   4 hours ago         Up 4 hours          0.0.0.0:80->80/tcp       docker-tutial
PS C:\Users\LenovoX\Downloads\app\app> docker-compose up
app_mysql_1 is up-to-date
app_app_1 is up-to-date
Attaching to app_mysql_1, app_app_1
mysql_1  | 2020-09-13 13:29:40+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 5.7.31-1debian10 started.
mysql_1  | 2020-09-13 13:29:41+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
mysql_1  | 2020-09-13 13:29:41+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 5.7.31-1debian10 started.
mysql_1  | 2020-09-13T13:29:41.429235Z 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
mysql_1  | 2020-09-13T13:29:41.510845Z 0 [Note] mysqld (mysqld 5.7.31) starting as process 1 ...
mysql_1  | 2020-09-13T13:29:41.536670Z 0 [Note] InnoDB: PUNCH HOLE support available
mysql_1  | 2020-09-13T13:29:41.536770Z 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mysql_1  | 2020-09-13T13:29:41.536782Z 0 [Note] InnoDB: Uses event mutexes
mysql_1  | 2020-09-13T13:29:41.536789Z 0 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
mysql_1  | 2020-09-13T13:29:41.536794Z 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
mysql_1  | 2020-09-13T13:29:41.536800Z 0 [Note] InnoDB: Using Linux native AIO
mysql_1  | 2020-09-13T13:29:41.538354Z 0 [Note] InnoDB: Number of pools: 1
mysql_1  | 2020-09-13T13:29:41.538589Z 0 [Note] InnoDB: Using CPU crc32 instructions
mysql_1  | 2020-09-13T13:29:41.543293Z 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
mysql_1  | 2020-09-13T13:29:41.555557Z 0 [Note] InnoDB: Completed initialization of buffer pool
mysql_1  | 2020-09-13T13:29:41.559932Z 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
mysql_1  | 2020-09-13T13:29:41.593050Z 0 [Note] InnoDB: Highest supported file format is Barracuda.
mysql_1  | 2020-09-13T13:29:41.772162Z 0 [Note] InnoDB: Creating shared tablespace for temporary tables
mysql_1  | 2020-09-13T13:29:41.772273Z 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
mysql_1  | 2020-09-13T13:29:42.117510Z 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
mysql_1  | 2020-09-13T13:29:42.118060Z 0 [Note] InnoDB: 96 redo rollback segment(s) found. 96 redo rollback segment(s) are active.
mysql_1  | 2020-09-13T13:29:42.118093Z 0 [Note] InnoDB: 32 non-redo rollback segment(s) are active.
mysql_1  | 2020-09-13T13:29:42.118816Z 0 [Note] InnoDB: 5.7.31 started; log sequence number 2720976
mysql_1  | 2020-09-13T13:29:42.119447Z 0 [Note] Plugin 'FEDERATED' is disabled.
mysql_1  | 2020-09-13T13:29:42.124676Z 0 [Note] Found ca.pem, server-cert.pem and server-key.pem in data directory. Trying to enable SSL support using them.
mysql_1  | 2020-09-13T13:29:42.124735Z 0 [Note] Skipping generation of SSL certificates as certificate files are present in data directory.
mysql_1  | 2020-09-13T13:29:42.125018Z 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
mysql_1  | 2020-09-13T13:29:42.125219Z 0 [Warning] CA certificate ca.pem is self signed.
mysql_1  | 2020-09-13T13:29:42.125273Z 0 [Note] Skipping generation of RSA key pair as key files are present in data directory.
mysql_1  | 2020-09-13T13:29:42.125838Z 0 [Note] Server hostname (bind-address): '*'; port: 3306
mysql_1  | 2020-09-13T13:29:42.125922Z 0 [Note] IPv6 is available.
mysql_1  | 2020-09-13T13:29:42.125935Z 0 [Note]   - '::' resolves to '::';
mysql_1  | 2020-09-13T13:29:42.125971Z 0 [Note] Server socket created on IP: '::'.
mysql_1  | 2020-09-13T13:29:42.127071Z 0 [Note] InnoDB: Buffer pool(s) load completed at 200913 13:29:42
mysql_1  | 2020-09-13T13:29:42.226932Z 0 [Warning] Insecure configuration for --pid-file: Location '/var/run/mysqld' in the path is accessible to all OS users. Consider choosing a different directory.
mysql_1  | 2020-09-13T13:29:42.258454Z 0 [Note] Event Scheduler: Loaded 0 events
mysql_1  | 2020-09-13T13:29:42.258925Z 0 [Note] mysqld: ready for connections.
mysql_1  | Version: '5.7.31'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  MySQL Community Server (GPL)
app_1    | yarn install v1.22.4
app_1    | [1/4] Resolving packages...
app_1    | success Already up-to-date.
app_1    | Done in 0.73s.
app_1    | yarn run v1.22.4
app_1    | $ nodemon src/index.js
app_1    | [nodemon] 1.19.2
app_1    | [nodemon] to restart at any time, enter `rs`
app_1    | [nodemon] watching dir(s): *.*
app_1    | [nodemon] starting `node src/index.js`
app_1    | Waiting for mysql:3306.
app_1    | Connected!
app_1    | Error: ER_HOST_NOT_PRIVILEGED: Host '172.19.0.2' is not allowed to connect to this MySQL server
app_1    |     at Handshake.Sequence._packetToError (/app/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
app_1    |     at Handshake.ErrorPacket (/app/node_modules/mysql/lib/protocol/sequences/Handshake.js:123:18)
app_1    |     at Protocol._parsePacket (/app/node_modules/mysql/lib/protocol/Protocol.js:291:23)
app_1    |     at Parser._parsePacket (/app/node_modules/mysql/lib/protocol/Parser.js:433:10)
app_1    |     at Parser.write (/app/node_modules/mysql/lib/protocol/Parser.js:43:10)
app_1    |     at Protocol.write (/app/node_modules/mysql/lib/protocol/Protocol.js:38:16)
app_1    |     at Socket.<anonymous> (/app/node_modules/mysql/lib/Connection.js:91:28)
app_1    |     at Socket.<anonymous> (/app/node_modules/mysql/lib/Connection.js:525:10)
app_1    |     at Socket.emit (events.js:315:20)
app_1    |     at addChunk (_stream_readable.js:295:12)
app_1    |     --------------------
app_1    |     at Protocol._enqueue (/app/node_modules/mysql/lib/protocol/Protocol.js:144:48)
app_1    |     at Protocol.handshake (/app/node_modules/mysql/lib/protocol/Protocol.js:51:23)
app_1    |     at PoolConnection.connect (/app/node_modules/mysql/lib/Connection.js:119:18)
app_1    |     at Pool.getConnection (/app/node_modules/mysql/lib/Pool.js:48:16)
app_1    |     at Pool.query (/app/node_modules/mysql/lib/Pool.js:202:8)
app_1    |     at /app/src/persistence/mysql.js:35:14
app_1    |     at new Promise (<anonymous>)
app_1    |     at Object.init (/app/src/persistence/mysql.js:34:12)
app_1    |     at processTicksAndRejections (internal/process/task_queues.js:97:5) {
app_1    |   code: 'ER_HOST_NOT_PRIVILEGED',
app_1    |   errno: 1130,
app_1    |   sqlMessage: "Host '172.19.0.2' is not allowed to connect to this MySQL server",
app_1    |   sqlState: undefined,
app_1    |   fatal: true
app_1    | }
app_1    | [nodemon] app crashed - waiting for file changes before starting...
mysql_1  | 2020-09-13T13:41:14.301693Z 0 [Note] Giving 0 client threads a chance to die gracefully
mysql_1  | 2020-09-13T13:41:14.301723Z 0 [Note] Shutting down slave threads
mysql_1  | 2020-09-13T13:41:14.301726Z 0 [Note] Forcefully disconnecting 0 remaining clients
mysql_1  | 2020-09-13T13:41:14.301732Z 0 [Note] Event Scheduler: Purging the queue. 0 events
mysql_1  | 2020-09-13T13:41:14.301787Z 0 [Note] Binlog end
mysql_1  | 2020-09-13T13:41:14.302329Z 0 [Note] Shutting down plugin 'ngram'
mysql_1  | 2020-09-13T13:41:14.302338Z 0 [Note] Shutting down plugin 'partition'
mysql_1  | 2020-09-13T13:41:14.302341Z 0 [Note] Shutting down plugin 'BLACKHOLE'
mysql_1  | 2020-09-13T13:41:14.302344Z 0 [Note] Shutting down plugin 'ARCHIVE'
mysql_1  | 2020-09-13T13:41:14.302346Z 0 [Note] Shutting down plugin 'PERFORMANCE_SCHEMA'
mysql_1  | 2020-09-13T13:41:14.302368Z 0 [Note] Shutting down plugin 'MRG_MYISAM'
mysql_1  | 2020-09-13T13:41:14.302371Z 0 [Note] Shutting down plugin 'MyISAM'
mysql_1  | 2020-09-13T13:41:14.302379Z 0 [Note] Shutting down plugin 'INNODB_SYS_VIRTUAL'
mysql_1  | 2020-09-13T13:41:14.302382Z 0 [Note] Shutting down plugin 'INNODB_SYS_DATAFILES'
mysql_1  | 2020-09-13T13:41:14.302384Z 0 [Note] Shutting down plugin 'INNODB_SYS_TABLESPACES'
mysql_1  | 2020-09-13T13:41:14.302385Z 0 [Note] Shutting down plugin 'INNODB_SYS_FOREIGN_COLS'
mysql_1  | 2020-09-13T13:41:14.302387Z 0 [Note] Shutting down plugin 'INNODB_SYS_FOREIGN'
mysql_1  | 2020-09-13T13:41:14.302389Z 0 [Note] Shutting down plugin 'INNODB_SYS_FIELDS'
mysql_1  | 2020-09-13T13:41:14.302391Z 0 [Note] Shutting down plugin 'INNODB_SYS_COLUMNS'
mysql_1  | 2020-09-13T13:41:14.302393Z 0 [Note] Shutting down plugin 'INNODB_SYS_INDEXES'
mysql_1  | 2020-09-13T13:41:14.302395Z 0 [Note] Shutting down plugin 'INNODB_SYS_TABLESTATS'
mysql_1  | 2020-09-13T13:41:14.302397Z 0 [Note] Shutting down plugin 'INNODB_SYS_TABLES'
mysql_1  | 2020-09-13T13:41:14.302398Z 0 [Note] Shutting down plugin 'INNODB_FT_INDEX_TABLE'
mysql_1  | 2020-09-13T13:41:14.302400Z 0 [Note] Shutting down plugin 'INNODB_FT_INDEX_CACHE'
mysql_1  | 2020-09-13T13:41:14.302402Z 0 [Note] Shutting down plugin 'INNODB_FT_CONFIG'
mysql_1  | 2020-09-13T13:41:14.302404Z 0 [Note] Shutting down plugin 'INNODB_FT_BEING_DELETED'
mysql_1  | 2020-09-13T13:41:14.302406Z 0 [Note] Shutting down plugin 'INNODB_FT_DELETED'
mysql_1  | 2020-09-13T13:41:14.302407Z 0 [Note] Shutting down plugin 'INNODB_FT_DEFAULT_STOPWORD'
mysql_1  | 2020-09-13T13:41:14.302409Z 0 [Note] Shutting down plugin 'INNODB_METRICS'
mysql_1  | 2020-09-13T13:41:14.302411Z 0 [Note] Shutting down plugin 'INNODB_TEMP_TABLE_INFO'
mysql_1  | 2020-09-13T13:41:14.302413Z 0 [Note] Shutting down plugin 'INNODB_BUFFER_POOL_STATS'
mysql_1  | 2020-09-13T13:41:14.302415Z 0 [Note] Shutting down plugin 'INNODB_BUFFER_PAGE_LRU'
mysql_1  | 2020-09-13T13:41:14.302417Z 0 [Note] Shutting down plugin 'INNODB_BUFFER_PAGE'
mysql_1  | 2020-09-13T13:41:14.302418Z 0 [Note] Shutting down plugin 'INNODB_CMP_PER_INDEX_RESET'
mysql_1  | 2020-09-13T13:41:14.302420Z 0 [Note] Shutting down plugin 'INNODB_CMP_PER_INDEX'
mysql_1  | 2020-09-13T13:41:14.302422Z 0 [Note] Shutting down plugin 'INNODB_CMPMEM_RESET'
mysql_1  | 2020-09-13T13:41:14.302424Z 0 [Note] Shutting down plugin 'INNODB_CMPMEM'
mysql_1  | 2020-09-13T13:41:14.302426Z 0 [Note] Shutting down plugin 'INNODB_CMP_RESET'
mysql_1  | 2020-09-13T13:41:14.302428Z 0 [Note] Shutting down plugin 'INNODB_CMP'
mysql_1  | 2020-09-13T13:41:14.302430Z 0 [Note] Shutting down plugin 'INNODB_LOCK_WAITS'
mysql_1  | 2020-09-13T13:41:14.302431Z 0 [Note] Shutting down plugin 'INNODB_LOCKS'
mysql_1  | 2020-09-13T13:41:14.302435Z 0 [Note] Shutting down plugin 'InnoDB'
mysql_1  | 2020-09-13T13:41:14.302487Z 0 [Note] InnoDB: FTS optimize thread exiting.
mysql_1  | 2020-09-13T13:41:14.402995Z 0 [Note] InnoDB: Dumping buffer pool(s) to /var/lib/mysql/ib_buffer_pool
mysql_1  | 2020-09-13T13:41:14.403476Z 0 [Note] InnoDB: Buffer pool(s) dump completed at 200913 13:41:14
mysql_1  | 2020-09-13T13:41:15.944238Z 0 [Note] InnoDB: Removed temporary tablespace data file: "ibtmp1"
mysql_1  | 2020-09-13T13:41:15.944423Z 0 [Note] Shutting down plugin 'MEMORY'
mysql_1  | 2020-09-13T13:41:15.944449Z 0 [Note] Shutting down plugin 'CSV'
mysql_1  | 2020-09-13T13:41:15.944463Z 0 [Note] Shutting down plugin 'sha256_password'
mysql_1  | 2020-09-13T13:41:15.944471Z 0 [Note] Shutting down plugin 'mysql_native_password'
mysql_1  | 2020-09-13T13:41:15.944920Z 0 [Note] Shutting down plugin 'binlog'
mysql_1  | 2020-09-13T13:41:15.946456Z 0 [Note] mysqld: Shutdown complete
mysql_1  |
app_app_1 exited with code 1
app_mysql_1 exited with code 0
PS C:\Users\LenovoX\Downloads\app\app> docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                NAMES
d78104b6cebd        docker101tutorial   "/docker-entrypoint.…"   4 hours ago         Up 4 hours          0.0.0.0:80->80/tcp   docker-tutial
PS C:\Users\LenovoX\Downloads\app\app> docker-compose up -d
app_app_1 is up-to-date
app_mysql_1 is up-to-date
PS C:\Users\LenovoX\Downloads\app\app> docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
39ce32aa8442        mysql:5.7           "docker-entrypoint.s…"   40 minutes ago      Up 28 minutes       3306/tcp, 33060/tcp      app_mysql_1
a2199bae223e        node:12-alpine      "docker-entrypoint.s…"   43 minutes ago      Up 28 minutes       0.0.0.0:3000->3000/tcp   app_app_1
d78104b6cebd        docker101tutorial   "/docker-entrypoint.…"   5 hours ago         Up 5 hours          0.0.0.0:80->80/tcp       docker-tutial
PS C:\Users\LenovoX\Downloads\app\app>

**Comment by Awawdi** [comment]
![130920201](https://user-images.githubusercontent.com/11771277/93020514-30e7c400-f5e6-11ea-9426-2bac84293558.PNG)

**Comment by mikesir87** [comment]
Hi @Awawdi and apologies for the super long delay getting to your issue.

It sounds like you might have been running into the issues related to the MySQL database not being found by the app. Unfortunately, the codebase (before yesterday) would silently hang waiting for DNS to resolve. Fortunately, that's been fixed and you should get a better indication when it's not working correctly.

Therefore, I'm going to close this issue. Feel free to open a new issue if you need any more assistance. Thanks!
